openmc/material.py

```diff
@@ -1181,22 +1172,38 @@ def export_to_xml(self, path='materials.xml'):
             Path to file to write. Defaults to 'materials.xml'.
 
         """
-
-        root_element = ET.Element("materials")
-        self._create_cross_sections_subelement(root_element)
-        self._create_material_subelements(root_element)
-
-        # Clean the indentation in the file to be user-readable
-        clean_indentation(root_element)
-
         # Check if path is a directory
         p = Path(path)
         if p.is_dir():
             p /= 'materials.xml'
 
-        # Write the XML Tree to the materials.xml file
-        tree = ET.ElementTree(root_element)
-        tree.write(str(p), xml_declaration=True, encoding='utf-8')
+        # Open the file in write mode.
```

Comment:
It would be useful to expand this comment a bit explaining why the file is being written in this manner as opposed to just taking the root element and writing all at once, as we do now.

Reply:
Good point! It's definitely helpful to document these subtle details